src/Symfony/Component/PropertyInfo/Util/PhpStanTypeHelper.php

```diff
@@ -128,7 +128,7 @@ private function extractTypes(TypeNode $node, NameScope $nameScope): array
             $collection = $mainType->isCollection() || \is_a($mainType->getClassName(), \Traversable::class, true) || \is_a($mainType->getClassName(), \ArrayAccess::class, true);
 
             // it's safer to fall back to other extractors if the generic type is too abstract
-            if (!$collection && !class_exists($mainType->getClassName())) {
+            if (!$collection && !class_exists($mainType->getClassName()) && !interface_exists($mainType->getClassName())) {
```

Comment:
```suggestion
            if (!$collection && !class_exists($mainType->getClassName()) && !interface_exists($mainType->getClassName(), false)) {
```

Reply:
Is this just microoptimization because `class_exists` already triggered autoloading and we dont need that for the second call of `interface_exists`?

Reply:
That's correct